client/txn.go

```diff
@@ -302,10 +310,11 @@ func endTxnCall(commit bool) proto.Call {
 func (txn *Txn) exec(retryable func(txn *Txn) error) (err error) {
 	// Run retryable in a retry loop until we encounter a success or
 	// error condition this loop isn't capable of handling.
+
 	for r := retry.Start(txn.db.txnRetryOptions); r.Next(); {
 		txn.haveTxnWrite, txn.haveEndTxn = false, false // always reset before [re]starting txn
 		if err = retryable(txn); err == nil {
-			if !txn.haveEndTxn && txn.haveTxnWrite {
+			if !txn.haveEndTxn && txn.txn.Writing {
```

Comment:
Ah ha I see you do have it in there. This does seem unnecessary. I'd really try hard to eliminate haveTxnWrite.